internal/txsubmitchecker/txsubmitchecker.go

```diff
@@ -15,126 +15,85 @@ import (
 	"github.com/neutron-org/neutron-query-relayer/internal/relay"
 )
 
-const TxCheckerNumWorkers = 4 // XXX: maybe this value should be configurable by user?
-
 var (
 	retryAttempts = retry.Attempts(4)
 	retryDelay    = retry.Delay(1 * time.Second)
 	retryError    = retry.LastErrorOnly(false)
 )
 
 type TxSubmitChecker struct {
-	inChan     <-chan relay.PendingSubmittedTxInfo
-	queue      chan relay.PendingSubmittedTxInfo
-	storage    relay.Storage
-	rpcClient  rpcclient.Client
-	logger     *zap.Logger
-	checkDelay time.Duration
+	storage   relay.Storage
+	rpcClient rpcclient.Client
+	logger    *zap.Logger
 }
 
 func NewTxSubmitChecker(
-	inQueue <-chan relay.PendingSubmittedTxInfo,
 	storage relay.Storage,
 	rpcClient rpcclient.Client,
 	logger *zap.Logger,
-	checkSubmittedTxStatusDelay uint64,
 ) *TxSubmitChecker {
 	return &TxSubmitChecker{
-		inQueue,
-		make(chan relay.PendingSubmittedTxInfo),
-		storage,
-		rpcClient,
-		logger,
-		time.Duration(checkSubmittedTxStatusDelay) * time.Second,
+		storage:   storage,
+		rpcClient: rpcClient,
+		logger:    logger,
 	}
 }
 
-func (tc *TxSubmitChecker) Run(ctx context.Context) error {
-	// we don't want to start jobs before we read all pending txs from database,
-	// hence we block on this read operation right in the beginning
+func (tc *TxSubmitChecker) Run(ctx context.Context, submittedTxsTasksQueue <-chan relay.PendingSubmittedTxInfo) error {
```

Comment:
I'm not sure this is good idea to move `submittedTxsTasksQueue` into Run(), cause we kinda give idea that you can use this function with multiple queues on the same object. But you're not, cause  it is tied to concrete `storage` that is in TxSubmitChecker struct

Reply:
not sure what your point is here; you can, for example, run multiple instances of `TxSubmitChecker` in separate goroutines to enable parallel processing of pending txs